Update code to be ready for NumPy 2.0

Our CI run with pre-release dependencies currently fail: https://github.com/quaquel/EMAworkbench/actions/runs/9154380335/job/25164846180

Which is caused by an update in NumPy 2.0 - which is currently in release-candidate phase.
```
FAILED test/test_analysis/test_prim.py::PrimBoxTestCase::test_inspect - AttributeError: `np.NaN` was removed in the NumPy 2.0 release. Use `np.nan` instead.. Did you mean: 'nan'?
```
Looks like a simple case, will look into it soon. We probably want to backport this.

--- a/ema_workbench/analysis/prim.py
+++ b/ema_workbench/analysis/prim.py
@@ -499,7 +499,7 @@ class PrimBox:
         for unc in uncs:
             values = self.box_lims[i][unc]
             box_lim.loc[unc] = values.values.tolist() + qp_values[unc]
-            box_lim.iloc[:, 2::] = box_lim.iloc[:, 2::].replace(-1, np.NaN)
+            box_lim.iloc[:, 2::] = box_lim.iloc[:, 2::].replace(-1, np.nan)
 
         return stats, box_lim
 
@@ -968,7 +968,7 @@ class PrimBox:
         # TODO:: this has knock on consequences
         # TODO:: elsewhere in the code (e.g. all box visualizations
         # TODO:: as well as in CART etc.)
-        # qp_values = qp_values.replace(-1, np.NaN)
+        # qp_values = qp_values.replace(-1, np.nan)
         qp_values = qp_values.to_dict(orient="list")
         return qp_values
 

